tests/integration/test_execute_code.py

```diff
@@ -49,6 +49,25 @@ def test_execute_python_code(random_code: str, random_string: str, agent: Agent)
     with open(destination) as f:
         assert f.read() == random_code
 
+def test_execute_python_code_filters_out_workspace_escaping(random_code: str, random_string: str, agent: Agent):
+    ai_name = agent.ai_name + "/../../"
+    result: str = sut.execute_python_code(random_code, "test_code", agent=agent)
```

Comment:
This doesn't actually test anything afaics, because the filename that is passed is `test_code`, not `../../test_code`